src/moleculesampleapp/CameraView.java

```diff
@@ -0,0 +1,197 @@
+
+        cameraScale = new Scale(1, 1, 1);
+
+        elements = new Group();
+
+        elements.getTransforms().addAll(xRotate, yRotate, zRotate, xyzTranslate);
+
+        cameraDistance = new Group(elements);
+
+        cameraDistance.getTransforms().addAll(planeTranslate, cameraScale);
+
+        camera = new PerspectiveCamera(false);
+
+        //camera.getTransforms().add(planeTranslate.createInverse());
+
+        cameraView = new Group(cameraDistance);
+
+        cameraView.getChildren().add(camera);
+
+        scene = new SubScene(cameraView, width, height, depthBuffer, sceneAntialiasing);
+
+        scene.setPickOnBounds(true); //to catch mouse also on "background"
+
+        scene.setCamera(camera);
+
+        setHandlers(scene);
+    }
+
+    public void setHandlers(SubScene subScene) {
+        subScene.setOnMousePressed(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent me) {
+                mousePosX = me.getSceneX();
+                mousePosY = me.getSceneY();
+                mouseOldX = me.getSceneX();
+                mouseOldY = me.getSceneY();
+
+                if (me.isMiddleButtonDown()) {
+
+                    offset.x = offset.y = offset.z = 0;
+                    scale = 0;
+                    angle.x = angle.y = angle.z = 0;
+
+                    xRotate.setAngle(Math.toDegrees(angle.x));
+                    yRotate.setAngle(Math.toDegrees(angle.y));
+                    zRotate.setAngle(Math.toDegrees(angle.z));
+
+                    xyzTranslate.setX(offset.x);
+                    xyzTranslate.setY(offset.y);
+                    xyzTranslate.setZ(offset.z);
+
+                    cameraScale.setX(Math.pow(10, scale));
+                    cameraScale.setY(Math.pow(10, scale));
+                    cameraScale.setZ(Math.pow(10, scale));
+                }
+            }
+        });
+        subScene.setOnMouseDragged(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent me) {
+                mouseOldX = mousePosX;
+                mouseOldY = mousePosY;
+                mousePosX = me.getSceneX();
+                mousePosY = me.getSceneY();
+                mouseDeltaX = (mousePosX - mouseOldX);
+                mouseDeltaY = (mousePosY - mouseOldY);
+
+                if (me.isControlDown()) {
+                    mouseDeltaX = 0;
+                }
+
+                if (me.isShiftDown()) {
+                    mouseDeltaY = 0;
+                }
+
+                if (me.isPrimaryButtonDown()) {
+                    Rotation baseRotation = new Rotation(RotationOrder.XYZ, angle.x, angle.y, angle.z);//get base rotation (it is transform form neutral point to actual point of view)
+                    Rotation deltaRotation = new Rotation(new Vector3D(0, 0), new Vector3D(-mouseDeltaX * ROTATE_SCALE, -mouseDeltaY * ROTATE_SCALE));//get rotation according to mouse movement
```

Comment:
[Extract local variables](https://www.jetbrains.com/idea/help/extract-variable.html) should solve problem.